File: laboratorio1.py
#!/usr/bin/python3

#Nombre y carne
    #Juan Diego Arce 
    #C20655
#Descripcion
    #El programa solicita un caracter y el tamaño de la base superior para asi formar un triángulo. 

def programa():
    caracter = str(input("Digite el caracter por usar: "))                             #El programa inicia solicitanto un caracter y lo transforma en str (string)  .                    
    base = int(input("Digite el tamano de la base del triángulo superior: "))          #Luego solicita la base superior del triángulo y lo transforma a int (entero).
    for b in range(1, base+1):                                                         #Utiliza "b" para obtener una repetición hasta base+1 (1, 2, 3, 4, 5, base+1).
        for p in range(1, b+1):                                                        #Utiliza "p" para obtener la misma repetición pero esta vez hasta b+1.
            print(caracter, end="")                                                    #Imprime el caracter elegido por el usuario "p" veces.
        print(" ")
    for b in range(base-1, 0, -1):                                                     #Luego se repite el proceso pero esta vez de manera inversa para crear el triángulo infeior.
        for p in range (1,b+1):                                                        #Para que en el triángulo inferior no se repita la base se usa el range desde (base-1).
            print(caracter, end="")
        print(" ")
    opcion = input("Digite SALIR para salir, cualquier cosa para continuar: ")
    if opcion == "SALIR":                                                              #Se crea un pequeño menú para que el usuario decida si seguir creando triágulos o si desea salirse.
        print("Saliendo...")
        print("Saliste")      
    else:
        programa()

programa()

File: laboratorio1.c
#include <stdio.h>

// Nombre: Juan Diego Arce
// Laboratorio 1

void imprimirPiramide(int tamanoBase, int tipo, char caracter) {
    int i;
    int j;

    if (tipo == 1) {
        // Pirámide normal
        for (i = 1; i <= tamanoBase; i++) {
            // Espacios en blanco
            for (j = 1; j <= tamanoBase - i; j++) {
                printf(" ");
            }
            // Caracter de la pirámide
            for (j = 1; j <= 2 * i - 1; j++) {
                printf("%c", caracter);
            }
            printf("\n");
        }
    } else {
        // Pirámide invertida
        for (i = tamanoBase; i >= 1; i--) {
            // Espacios en blanco
            for (j = 1; j <= tamanoBase - i; j++) {
                printf(" ");
            }
            // Caracter de la pirámide
            for (j = 1; j <= 2 * i - 1; j++) {
                printf("%c", caracter);
            }
            printf("\n");
        }
    }
}

int main() {
    int tamanoBase, tipo;
    char caracter;
    char continuar;
    do {
        // Solicita al usuario el tamaño de la pirámide, el caracter a utilizar y el tipo de pirámide
        printf("\nIngrese el tamaño de la base de la pirámide: ");
        scanf("%d", &tamanoBase);

        printf("Ingrese el caracter que desea utilizar: ");
        scanf(" %c", &caracter);

        printf("\n1: Pirámide Normal \n2: Pirámide Invertida \n \nIngrese el tipo de pirámide que desea imprimir: ");
        scanf("%d", &tipo);

        // Imprime la pirámide
        imprimirPiramide(tamanoBase, tipo, caracter);

        // Pregunta si desea continuar
        printf("\nDesea continuar imprimiendo pirámides? \n\nNo: SALIR \nSi: Y\n\nDigite su respuesta: ");
        scanf(" %c", &continuar);

    } while (continuar == 'y' || continuar == 'Y');

    return 0;
}
